cmd/minikube/cmd/delete.go

```diff
@@ -46,10 +48,43 @@ associated files.`,
 
 // runDelete handles the executes the flow of "minikube delete"
 func runDelete(cmd *cobra.Command, args []string) {
-	if len(args) > 0 {
-		exit.UsageT("usage: minikube delete")
+	profileFlag, err := cmd.Flags().GetString("profile")
+	if err != nil {
+		exit.WithError("Could not get profile flag", err)
 	}
-	profile := viper.GetString(pkg_config.MachineProfile)
+
+	if deleteAll {
+		if profileFlag != constants.DefaultMachineName {
+			exit.UsageT("usage: minikube delete --all")
+		}
+
+		validProfiles, invalidProfiles, err := pkg_config.ListProfiles()
+		profilesToDelete := append(validProfiles, invalidProfiles...)
+
+		if err != nil {
+			exit.WithError("Error getting profiles to delete", err)
+		}
+
+		deleteAllProfiles(profilesToDelete)
+	} else {
+		if len(args) > 0 {
+			exit.UsageT("usage: minikube delete")
+		}
+
+		profileName := viper.GetString(pkg_config.MachineProfile)
+		profile, err := pkg_config.LoadProfile(profileName)
+
+		if err != nil {
+			exit.WithError("Could not load profile", err)
+		}
+		deleteProfile(profile)
+	}
+}
+
+//TODO Return errors and handle in runDelete?
+func deleteProfile(profile *pkg_config.Profile) {
```

Comment:
please add unit tests simmilar to this by making dummy profiles with valid and invalid profiles in testData and make sure it deletes them:
https://github.com/kubernetes/minikube/tree/master/pkg/minikube/config/testdata/.minikube/profiles

Reply:
Do you think the profile Json-Files are enough to test or should we also add testdata for the machine?

Reply:
I rather separate testdata for delete test. So we can run the tests in parallel. So if test deletes the folders it doesn't affect the other tests. It's cheap to add a few json files.

Reply:
Feel free to copy the same testdata for profile or make your own.